compiler/rustc_parse/src/parser/path.rs

```diff
@@ -953,6 +966,24 @@ impl<'a> Parser<'a> {
                 }
             }
         };
+
+        if let Some(attr_span) = attr_span {
+            let guar = self.dcx().emit_err(AttributeOnGenericType {
+                span: attr_span,
+                fix_span: attr_span.until(arg.span()),
+            });
+
+            return Ok(Some(match arg {
+                GenericArg::Type(_) => GenericArg::Type(self.mk_ty(attr_span, TyKind::Err(guar))),
+                GenericArg::Const(_) => {
+                    // Create error const expression
+                    let error_expr = self.mk_expr(attr_span, ExprKind::Err(guar));
+                    GenericArg::Const(AnonConst { id: ast::DUMMY_NODE_ID, value: error_expr })
+                }
+                GenericArg::Lifetime(lt) => GenericArg::Lifetime(lt), // Less common case
+            }));
```

Comment:
I don't think we need to *taint* or *poison* the arg just because it has attrs. Emitting an error is already sufficient at guaranteeing a failed compilation. This whole `return` could be removed.